.github/CONTRIBUTING.md

```diff
@@ -129,19 +129,20 @@ Additional references:
   If the changes are related to an existing GitHub issue,
   please reference the issue in PR description (e.g. ```Fix #11```).
   See [this][closing-via-message] for more details.
-* If the change warrants a note in the [changelog](../CHANGELOG.MD)
-  either update the changelog in your pull request or
-  add a comment in the PR description saying that the change may warrant a note in the changelog.
-  New changes always go into the **Unreleased** section.
+
+#### Pull request - Change log
+
+* All PRs must update the [changelog](../CHANGELOG.MD) in your pull request.
```

Comment:
Based on previous experience I don't like the requirement.
 - We reject formatting changes to get git commit that only contain significant changes. With this requirement, we will have trash in git commit. There is now a need to move commits to other branches. There will be more of these jobs over time. It's harder to do with this trash.
- This will slow down the performance of community users and irritate them.

Finally, we can easily automate it. I previously sent @daxian-dbw  a sample script. If we'll to keep the PR headers neat, we'll get a neat log of the automatic way.

Reply:
We reject changes which contain both formatting and functional changes to the same file.  This is not the same thing at all.  
Unless we say, maintainers are not part of the community, I don't think we can say this.  The PowerShell DSC community and other communities I've participated in have this requirement without issue.

Reply:
https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#fixing-an-issue  Item number 7.

Reply:
Finally, I'm not sure it's easy to keep the PR `headers` clean and therefore automate it.  By header, I assume you mean the message, and the message is often too limited to use for as the changelog.  If you mean the notes, since they are unstructured, they suffer other problems.  

If the author refuses to do this, the maintainer can always update the changelog for them, but in the DSC community, this was rarely an issue.

Reply:
DSC repository has less than 100 PR in the last two years. We have so much in two months. We will actually have a conflict in every PR with this file.

Previously, the team required to update the file and then forget it. Why? Probably because it's not a functional change that everyone forgets and ignores.

I'm sorry, I don't understand the usefulness of this file while there is an open GitHub - any consumer can at any time not only look at the history of changes, but also the history of the code.

This is a secondary element to me. If this element requires so much effort, I immediately think that it should be removed or automated. If we can't use a PR header/description, we could think about using a commit header/description.

Reply:
There is an attribute on the file to use a Union merge, which is supposed to prevent merge conflicts.  This may cause some issues in the file, but should produce a usable file.  Many other projects use this method.

When you were looking at the DSC Repository, did you count all the submodules?  I asked the DSC team and DSC had 43 PRs in November across all the repos.

As to the purpose of the file, most people that just want to consume the project would like a simplified way of viewing what has been changed in the project from their perspective.  This is not what the PR or the commit log is about and this needs to be reviewed in context to make sure it's being expressed clearly.  For example, #2253 suggest we include a link to the changelog in the What's New section of the documentation.

Even if it were automated, translating from the contributor focused messaged of the PR to the PowerShell user-focused message in the changelog is what is taking the time.  Automating gathering the PR descriptions and links would only solve at best half the problem.

Reply:
>When you were looking at the DSC Repository, did you count all the submodules? I asked the DSC team and DSC had 43 PRs in November across all the repos.

No, I follow your link. I wonder do they have one changelog for all repos? If no they never have merge conflicts as we had previously with the file.

I'm not the first person who wants this "kill" feature:
https://pypi.python.org/pypi/gitchangelog#changelog
https://www.npmjs.com/package/auto-changelog
Also I found [this project](https://github.com/semantic-release/semantic-release#generatenotes) very interesting (and links there too).

My comments isn't blocking.

Reply:
I think this one is the most interesting:  https://github.com/skywinder/github-changelog-generator    But, it needs a LOT of work to be usable.  This issue needs to be fixed, and we would like to be able to group by tags.

Reply:
Tracking Issue #5614